src/components/InfiniteGrids/useInfiniteGrid.ts

```diff
@@ -130,7 +132,12 @@ export const useInfiniteGrid = <
     return () => window.removeEventListener('scroll', scrollHandler)
   }, [isReady, loading, allItemsLoaded, onScrollToBottom, onDemand])
 
-  const displayedEdges = data?.edges.slice(skipCount, targetLoadedItemsCount) ?? []
+  const edges = sortByViews
+    ? [...(data?.edges || [])].sort((a, b) => {
+        return ((b?.node as ChannelEdge['node']).views || 0) - ((a?.node as ChannelEdge['node']).views || 0)
+      })
+    : data?.edges
```

Comment:
1. This needs to be split somehow, it's not readable at the moment
2. You assume `ChannelEdge` but this will handle videos as well

Reply:
I changed the approach a bit, I replaced the `sortByViews` boolean with the sort function(`additionalSortFn`). It will make `useInfniteGrid` more flexible I think. Still not sure how to better type this, but it should be more readable now.